deepmd/utils/data_system.py

```diff
@@ -375,19 +383,88 @@ def get_batch(self, sys_idx: Optional[int] = None):
             The index of system from which the batch is get.
             If sys_idx is not None, `sys_probs` and `auto_prob_style` are ignored
             If sys_idx is None, automatically determine the system according to `sys_probs` or `auto_prob_style`, see the following.
+            This option does not work for mixed systems.
+
+        Returns
+        -------
+        dict
+            The batch data
         """
         if not hasattr(self, "default_mesh"):
             self._make_default_mesh()
-        if sys_idx is not None:
-            self.pick_idx = sys_idx
+        if not self.mixed_systems:
+            if sys_idx is not None:
+                self.pick_idx = sys_idx
+            else:
+                # prob = self._get_sys_probs(sys_probs, auto_prob_style)
+                self.pick_idx = dp_random.choice(
+                    np.arange(self.nsystems), p=self.sys_probs
+                )
+            b_data = self.data_systems[self.pick_idx].get_batch(
+                self.batch_size[self.pick_idx]
+            )
+            b_data["natoms_vec"] = self.natoms_vec[self.pick_idx]
+            b_data["default_mesh"] = self.default_mesh[self.pick_idx]
         else:
-            # prob = self._get_sys_probs(sys_probs, auto_prob_style)
-            self.pick_idx = dp_random.choice(np.arange(self.nsystems), p=self.sys_probs)
-        b_data = self.data_systems[self.pick_idx].get_batch(
-            self.batch_size[self.pick_idx]
-        )
-        b_data["natoms_vec"] = self.natoms_vec[self.pick_idx]
-        b_data["default_mesh"] = self.default_mesh[self.pick_idx]
```

Comment:
better to write two functions for get_batch (standard) and get_batch (mixed) and switch between the functions in the if-else condition.